R/readNWISunit.r

```diff
@@ -472,4 +480,77 @@ readNWISstat <- function(siteNumbers, parameterCd, startDate = "", endDate = "",
+#' of Columbia is also available.  Default is \code{NULL}, which will return national data.
+#' @param county character Name(s) of counties.  Default is \code{NULL}, which will return state or national data 
+#' depending on the stateAB argument.  \code{ALL} may also be supplied, which will return data for every county in a 
+#' state. County names be supplied ending in "County" or not — e.g., "Belmont" and "Belmont County" are both acceptable.  
+#' @param convertType logical defaults to \code{TRUE}. If \code{TRUE}, the function will convert the data to
+#' numerics based on a standard algorithm. Years, months, and days (if appliccable) are also returned as numerics
+#' in separate columns.  If convertType is false, everything is returned as a character.
+#' @return A data frame with at least the year of record, and all available statistics for the given geographic parameters.
+#' County and state fields will be included as appropriate.
+#' 
+#' @export
+#' @importFrom stringr str_sub
+#' @examples 
+#' \dontrun{
+#' #All data for a county
+#' allegheny <- readNWISuse(years="ALL",stateAB = "PA",county = "Allegheny")
+#' 
+#' #Data for an entire state for certain years
+#' ohio <- readNWISuse(years=c(2000,2005,2010),stateAB = "OH")
+#' 
+#' #Data for an entire state, county by county
+#' pr <- readNWISuse(years=c(2000,2005,2010),stateAB = "PR",county="ALL")
+#' 
+#' #All national-scale data
+#' national <- readNWISuse()
+#' 
+#' #Washington, DC data
+#' dc <- readNWISuse(stateAB = "DC")
+#' 
+#' }
+ 
+readNWISuse <- function(years="ALL",stateAB=NULL,county=NULL,convertType=TRUE){
+  if(!is.null(county) && is.null(stateAB)){
+    stop("Please enter a state code")
+  }
+ 
+  if(!is.null(county) && toupper(county) != "ALL"){
+    #check if "County" was included on string - need it to match countyCd data frame
+    county <- ifelse(!grepl('\\County$',county),paste(county,"County"),county)
+    
+    #county turns into number for the particular state if isn't empty or ALL
+    stateCounties <- countyCd[countyCd$STUSAB==stateAB,]
+    county <- paste(stateCounties[match(county,stateCounties$COUNTY_NAME),]$COUNTY,collapse="%2C")
+    if(length(county)==0){
+      stop("There was a problem with county; check that your entry matches the countyCd data frame")
+    }
+  }
+  if(!is.null(county) && toupper(county) == "ALL"){county <- toupper(county)} #case sensitive in URL
+  if(any(grepl("(?i)all",years))){years <- toupper(years)}
+  url <- constructUseURL(years,stateAB,county)
+  data <- importRDB1(url,convertType=convertType)  
+  
+  #distinguish if total country data arriving in named rows
+  if(is.null(county) && is.null(stateAB)){
+    cmmnt <- comment(data)
+    data <- t(data)
+    colnames(data) <- data[1,]
+    data <- as.data.frame(data[-1,],stringsAsFactors=FALSE)
+    data <- cbind(Year=as.integer(str_sub(rownames(data),2)),data)
```

Comment:
yeah, this will cause errors without adding a `stringr` as a new dependency. In general, I've been trying to limit the dependencies as much as possible. Let's see if we can do it in base